smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/PaginatorGenerator.kt

```diff
@@ -0,0 +1,245 @@
+        writer
+            .addImport(ExternalTypes.KotlinxCoroutines.Flow)
+            .addImport(ExternalTypes.KotlinxCoroutines.FlowGenerator)
+            .addImport(serviceSymbol)
+            .addImport(inputSymbol)
+            .addImport(outputSymbol)
+            .addImport(cursorSymbol)
+            .addImportReferences(cursorSymbol, SymbolReference.ContextOption.DECLARE)
+            .withBlock(
+                "fun #T.#LPaginated(initialRequest: #T): Flow<#T> =",
+                "",
+                serviceSymbol,
+                operationShape.defaultName(),
+                inputSymbol,
+                outputSymbol
+            ) {
+                withBlock("flow {", "}") {
+                    write("var cursor: #F = null", cursorSymbol)
+                    write("var isFirstPage: Boolean = true")
+                    write("")
+                    withBlock("while (isFirstPage || (cursor?.isNotEmpty() == true)) {", "}") {
+                        withBlock("val req = initialRequest.copy {", "}") {
+                            write("this.$markerLiteral = cursor")
+                        }
+                        write(
+                            "val result = this@#1LPaginated.#1L(req)",
+                            operationShape.defaultName()
+                        )
+                        write("isFirstPage = false")
+                        write("cursor = result.$nextMarkerLiteral")
+                        write("emit(result)")
+                    }
+                }
+            }
+    }
+
+    // Generate a paginator that iterates over the model-specified item
+    private fun renderItemPaginator(
+        writer: KotlinWriter,
+        serviceShape: ServiceShape,
+        operationShape: OperationShape,
+        itemDesc: ItemDescriptor,
+        outputSymbol: Symbol,
+    ) {
+        writer.write("")
+        writer.dokka(
+            """
+                 This paginator transforms the flow returned by [${operationShape.defaultName()}Paginated] to access the nested member [${itemDesc.targetMember.defaultName(serviceShape)}]
+                 @return a [kotlinx.coroutines.flow.Flow] that can collect [${itemDesc.targetMember.defaultName(serviceShape)}]
+            """.trimIndent()
+        )
+        writer.dokka("Paginate over [${outputSymbol.name}.${itemDesc.itemLiteral}]")
+        writer
+            .addImport(ExternalTypes.KotlinxCoroutines.FlowTransform)
+            .addImport(itemDesc.itemSymbol)
+            .addImportReferences(itemDesc.itemSymbol, SymbolReference.ContextOption.USE)
+            // @JvmName is required due to Java interop compatibility in the compiler.
+            // Multiple functions may have the same name and the generic does not disambiguate the type in Java.
+            // NOTE: This does not mean these functions are callable from Java.
+            .write(
+                """@JvmName("#L#L")""",
+                outputSymbol.name.toggleFirstCharacterCase(),
```

Comment:
**Suggestion**: Explicitly casting the first character to lowercase would clarify the intent here.

Reply:
This?

```
outputSymbol.name.replaceFirstChar {
    if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString()
}
```

Reply:
I'm confused...do we _want_ this to be uppercase? That's not normally how Java/Kotlin methods are named. Assuming not, I think this captures the intent more clearly:

```kotlin
outputSymbol.name.replaceFirstChar(Char::lowercaseChar)
```